freqtrade/exchange/binance.py

```diff
@@ -98,38 +102,74 @@ def additional_exchange_init(self) -> None:
         except ccxt.BaseError as e:
             raise OperationalException(e) from e
 
-    async def _async_get_historic_ohlcv(
+    def get_historic_ohlcv(
         self,
         pair: str,
         timeframe: str,
         since_ms: int,
         candle_type: CandleType,
         is_new_pair: bool = False,
-        raise_: bool = False,
         until_ms: Optional[int] = None,
-    ) -> OHLCVResponse:
+    ) -> DataFrame:
         """
         Overwrite to introduce "fast new pair" functionality by detecting the pair's listing date
         Does not work for other exchanges, which don't return the earliest data when called with "0"
         :param candle_type: Any of the enum CandleType (must match trading mode!)
         """
         if is_new_pair:
-            x = await self._async_get_candle_history(pair, timeframe, candle_type, 0)
```

Comment:
I think we'll want to have the new mode optional (but on by default!).

Implicitly, this makes the previous override of "get the oldest available data" necessary again.

I'm not entirely sure how availability of the binance.vision page is across the world - so by forcing this onto users, we may break data-download for them completely if their location isn't able to access binance.vision.

Reply:
As a side note - is this (get the oldest possible datapoint) handled (or even necessary) in the new mode? like - if i download with 20170101- - but the pair was listed in 2023 ... i'd be making 1000nds of calls to get daily files ... while in reality - i'd only need 300 ...

Reply:
>Implicitly, this makes the previous override of "get the oldest available data" necessary again.

I did not remove this part, but moved it to `get_historic_ohlcv` so that both modes can utilize it.

>is this (get the oldest possible datapoint) handled (or even necessary) in the new mode?

Yes, as said above.

Reply:
- Added a paramter `only_from_ccxt` in `Exchange.get_historic_ohlcv` which defaults to `False`, we can pass `True` to disable the new mode.
- Stripped the new mode into an isolated method `Binance.get_historic_ohlcv_fast`, the new mode can be easily eliminated when we need.

[refactor: strip fast download logic into a method](https://github.com/freqtrade/freqtrade/pull/10858/commits/acc53065e5fa7ab5197073276306dc9dc3adbfa3)